src/test/java/org/assertj/core/api/SoftAssertionsTest.java

```diff
@@ -151,6 +151,13 @@ public void setup() {
     iterableMap.put("rank", ranks);
   }
 
+  @Test
+  public void should_collect_soft_check_assertion() {
+    softly.check(() -> assertThat(true).isFalse());
+    softly.check(() -> assertThat(true).isTrue());
+    assertThat(softly.errorsCollected()).hasSize(1);
+  }
+
```

Comment:
Add a test when the lambda throws an exception and check it is propagated as a runtime exception.